src/wp-includes/fonts/fonts.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Fonts functions.
+ *
+ * @package    WordPress
+ * @subpackage Fonts
+ * @since      6.4.0
+ */
+
+/**
+ * Generates and prints font-face styles for given fonts or theme.json fonts.
+ *
+ * @since 6.4.0
+ *
+ * @param array[][] $fonts {
+ *     Optional. The font-families and their font variations. Default empty array.
+ *
+ *     @type string $font-family => array[] $variations {
+ *         Optional. An associated array of font variations for this font-family.
+ *         Each variation has the following structure.
+ *
+ *         @type array $font_variation {
+ *             @type string          $font-family             The font-family property.
+ *             @type string|string[] $src                     The URL(s) to each resource containing the font data.
+ *             @type string          $font_style              Optional. The font-style property. Default 'normal'.
+ *             @type string          $font-weight             Optional. The font-weight property. Default '400'.
+ *             @type string          $font-display            Optional. The font-display property. Default 'fallback'.
+ *             The following font-face properties are optional and will not be generated if not provided.
+ *             @type string          $ascent-override         Optional. The ascent-override property.
+ *             @type string          $descent-override        Optional. The descent-override property.
+ *             @type string          $font-stretch            Optional. The font-stretch property.
+ *             @type string          $font-variant            Optional. The font-variant property.
+ *             @type string          $font-feature-settings   Optional. The font-feature-settings property.
+ *             @type string          $font-variation-settings Optional. The font-variation-settings property.
+ *             @type string          $line-gap-override       Optional. The line-gap-override property.
+ *             @type string          $size-adjust             Optional. The size-adjust property.
+ *             @type string          $unicode-range           Optional. The unicode-range property.
+ *         }
+ *     }
+ * }
+ */
+function wp_print_font_faces( $fonts = array() ) {
+	static $wp_font_face = null;
+
+	if ( empty( $fonts ) ) {
+		$fonts = WP_Font_Face_Resolver::get_fonts_from_theme_json();
+	}
+
+	if ( empty( $fonts ) ) {
+		return;
+	}
+
+	if (
+		null === $wp_font_face ||
+
+		// Ignore cache when automated test suites are running.
+		( defined( 'WP_RUN_CORE_TESTS' ) && WP_RUN_CORE_TESTS )
```

Comment:
The intent:

* Use a new instance of `WP_Font_Face` for each PHPUnit test. Why? To avoid test-to-test interactions by ensuring a new object is used for each test.

* Else, it acts as a singleton whether in `WP_DEBUG`, prod, or development modes.

Reply:
Do the tests pass with a combination of `@runInSeparateProcess` and `@preserveGlobalState disabled` in the test docblocks?

If so, while it would slow down the tests, it removes the need for this test-related source code and separates implemention from the test suite. It may be worth considering if it hasn't been already.

Reply:
All of the Font Face tests would require running in their own separate processes. What would the test performance impact be?

* The code now: 00:05.447
* Running in separate processes: 00:36.257

That's fairly significant for 20 tests. Ouch.

Reply:
Whoopsie, I was wrong. Only the global function tests (i.e. for `wp_print_font_faces()`) need to run in a separate process, not all of the tests.

Re-running the `fonts` group tests with the `wp_print_font_faces()` tests in separate processes: 00:16.807

Adds about 11 seconds with the current tests, i.e. more unhappy tests are coming.

What do you think @costdev?

Reply:
Colin and I chatted in Slack. Commit https://github.com/WordPress/wordpress-develop/pull/5051 runs the tests in separate processes. 

Results: adds ~11 seconds to the run time of the test suites 😢 , but removes the test constant from the source code 😄 .